packages/next/src/server/app-render/app-render.tsx

```diff
@@ -877,6 +877,8 @@ async function renderToHTMLOrFlightImpl(
     res,
   }
 
+  getTracer().getRootSpanAttributes()?.set('next.route', pagePath)
```

Comment:
Why move it up? Are some steps below sensitive to it?

Overall, the `pagePath` is not a great value here and it takes some time to update it to the right value. IMHO, too late.

Reply:
> Why move it up? Are some steps below sensitive to it?

Yes - the `if (isRSCRequest && !isStaticGeneration) { ... }` check is where this function can end early, which affects whether `next.route` gets set. 

But also, the snapshot testing looks problematic, for example this is from the most recent PR build-and-test logs:

![image](https://github.com/vercel/next.js/assets/967811/b62d27f0-72ac-4cbc-b5d6-9f6b226c4267)

id, parentId, timestamp, traceId & duration values appear dependent on something that's preventing consistent results. Maybe [jest.setSystemTime[(https://jestjs.io/docs/jest-object#jestsetsystemtimenow-number--date) could address the timestamp part? For the others I'm not sure what might ensure that consistency.

Reply:
We can definitely just use `matchAny` for timestamp and duration. And probably for traceId. However, id and parentId are not so simple - they represent the nesting hierarchy which is somewhat important for telemetry. We can matchAny them too, but maybe we can find a way to link them up? E.g. maybe we can drop id/parentId, but instead add a synthetic parentName? That way we can test hierarchy as well.